src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -186,6 +208,18 @@ namespace QgsWms
     QgsLegendSettings settings = legendSettings();
     QgsLayerTreeModelLegendNode::ItemContext ctx;
     ctx.painter = painter.get();
+
+    // create context
+    QgsRenderContext context = QgsRenderContext::fromQPainter( painter.get() );
```

Comment:
This block is identical to the one above?
Can it be deduplicated?

Reply:
I thought about that. Would you make a new method just for that? I decided against it, but would be fine for me.

Reply:
There's a certain risk of forgetting upgrades in the future. I would.